Common/Orders/Fills/EquityFillModel.cs

```diff
@@ -13,204 +13,174 @@
-                    throw new ArgumentOutOfRangeException();
-            }
-            return new Fill(orderEvent);
-        }
-
-        /// <summary>
-        /// Default market fill model for the base security class. Fills at the last traded price.
+        /// Market fill model for the Equity. Fills at the last bid or ask price.
         /// </summary>
         /// <param name="asset">Security asset we're filling</param>
         /// <param name="order">Order packet to model</param>
         /// <returns>Order fill information detailing the average price and quantity filled.</returns>
-        public virtual OrderEvent MarketFill(Security asset, MarketOrder order)
+        public override OrderEvent MarketFill(Security asset, MarketOrder order)
         {
-            //Default order event to return.
+            // Default order event to return.
             var utcTime = asset.LocalTime.ConvertToUtc(asset.Exchange.TimeZone);
             var fill = new OrderEvent(order, utcTime, OrderFee.Zero);
 
-            if (order.Status == OrderStatus.Canceled) return fill;
-
-            // make sure the exchange is open/normal market hours before filling
-            if (!IsExchangeOpen(asset, false)) return fill;
-
-            var prices = GetPricesCheckingPythonWrapper(asset, order.Direction);
-            var pricesEndTimeUtc = prices.EndTime.ConvertToUtc(asset.Exchange.TimeZone);
-
-            // if the order is filled on stale (fill-forward) data, set a warning message on the order event
-            if (pricesEndTimeUtc.Add(Parameters.StalePriceTimeSpan) < order.Time)
+            if (order.Status == OrderStatus.Canceled ||            // Orders with canceled status or hold
+                order.Direction == OrderDirection.Hold ||          // direction don't need anymore checks
+                !IsExchangeOpen(asset, false))   // Exchange need to be open/normal market hours
             {
-                fill.Message = $"Warning: fill at stale price ({prices.EndTime.ToStringInvariant()} {asset.Exchange.TimeZone})";
+                return fill;
             }
 
-            //Order [fill]price for a market order model is the current security price
-            fill.FillPrice = prices.Current;
-            fill.Status = OrderStatus.Filled;
-
-            //Calculate the model slippage: e.g. 0.01c
+            // Calculate the model slippage: e.g. 0.01c
             var slip = asset.SlippageModel.GetSlippageApproximation(asset, order);
 
-            //Apply slippage
+            // Define the last bid or ask time to set stale prices message
+            var endTime = DateTime.MinValue;
+
+            // Apply slippage
             switch (order.Direction)
             {
                 case OrderDirection.Buy:
-                    fill.FillPrice += slip;
+                    //Order [fill]price for a buy market order model is the current security ask price
+                    fill.FillPrice = GetAskPrice(asset, out endTime) + slip;
```

Comment:
We should probably add this change to the PR description too: market orders will now use bid/ask versus ...?